File: playground/python/infrastructure/repositories/order_repository.py
"""Order repository adapter."""

from __future__ import annotations

from playground.python.domain.entities.order import Order
from playground.python.domain.value_objects.order_id import OrderId


class OrderRepository:
    """In-memory order repository (infrastructure adapter)."""

    def __init__(self) -> None:
        self._orders: dict[str, Order] = {}

    def save(self, order: Order) -> None:
        """Persist an order."""
        self._orders[order.order_id.value] = order

    def find_by_id(self, order_id: OrderId) -> Order | None:
        """Find order by id."""
        return self._orders.get(order_id.value)

File: playground/typescript/src/infrastructure/repositories/OrderRepository.ts
import type { Order } from "../../domain/entities/Order.js";
import type { OrderId } from "../../domain/value_objects/OrderId.js";

/** In-memory order repository. */
export class OrderRepository {
  private readonly orders = new Map<string, Order>();

  save(order: Order): void {
    this.orders.set(order.orderId.value, order);
  }

  findById(orderId: OrderId): Order | null {
    return this.orders.get(orderId.value) ?? null;
  }
}
